platform/src/main/java/org/hillview/storage/GenericLogs.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2017 MapleLabs. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hillview.storage;
+
+import org.apache.commons.lang.StringEscapeUtils;
+import org.hillview.table.ColumnDescription;
+import org.hillview.table.Schema;
+import org.hillview.table.Table;
+import org.hillview.table.api.ContentsKind;
+import org.hillview.table.api.ITable;
+import org.hillview.storage.FileSetDescription;
+import org.hillview.utils.Utilities;
+
+import java.util.Map;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+
+import io.krakens.grok.api.*;
+
+/**
+ * Reads Generic logs into ITable objects.
+ */
+
+public class GenericLogs {
+    private static Schema schema;
+    private static int schema_columns;
+    private static Grok grok;
+    private static String logFormat;
+
+    GenericLogs(String logFormat) {
+        this.logFormat = logFormat;
+        GrokCompiler grokCompiler = GrokCompiler.newInstance();
+        grokCompiler.registerDefaultPatterns();
+        grokCompiler.registerPatternFromClasspath("/patterns/log-patterns");
+        grok = grokCompiler.compile(this.logFormat, true);
+        schema_columns = 0;
+        schema = new Schema();
+        this.schema.append(new ColumnDescription("Host", ContentsKind.String));
+    }
+
+    public static class LogFileLoader extends TextFileLoader {
+        LogFileLoader(final String path) {
+            super(path);
+        }
+
+        void parse(String line, String[] output) {
+            Match gm = grok.match(line);
+            final Map<String, Object> capture = gm.capture();
+            if (capture.size() > 0) {
+                output[0] = Utilities.getHostName();
```

Comment:
You should probably save the hostname in a field - this function may be very expensive and it always returns the same thing.